samples/startup-script/index.js

```diff
@@ -26,110 +26,114 @@ const zone = compute.zone('us-central1-a');
+        os: 'ubuntu',
+        http: true,
+        metadata: {
+          items: [
+            {
+              key: 'startup-script',
+              value: `#! /bin/bash
+    
+            # Installs apache and a custom homepage
+            apt-get update
+            apt-get install -y apache2
+            cat <<EOF > /var/www/html/index.html
+            <!doctype html>
+            <h1>Hello World</h1>
+            <p>This page was created from a simple start-up script!</p>`,
+            },
+          ],
         },
-      ],
-    },
-  };
-
-  const vm = zone.vm(name);
-
-  vm.create(config)
-    .then(data => {
-      const operation = data[1];
-      return operation.promise();
-    })
-    .then(() => {
-      return vm.getMetadata();
-    })
-    .then(data => {
-      const metadata = data[0];
-
+      };
+      const vm = zone.vm(name);
+      
+      const data = await vm.create(config);
+      console.log('Creating VM ...');
+      await data[1].promise();
+      const metadata = await vm.getMetadata();
+      
       // External IP of the VM.
-      const ip = metadata.networkInterfaces[0].accessConfigs[0].natIP;
+      const ip = metadata[0].networkInterfaces[0].accessConfigs[0].natIP;
       console.log(`Booting new VM with IP http://${ip}...`);
 
       // Ping the VM to determine when the HTTP server is ready.
       let waiting = true;
-      const timer = setInterval(
-        ip => {
-          http
-            .get('http://' + ip, res => {
-              const statusCode = res.statusCode;
-              if (statusCode === 200 && waiting) {
-                waiting = false;
-                clearTimeout(timer);
-                // HTTP server is ready.
-                console.log('Ready!');
-                callback(null, ip);
-              }
-            })
-            .on('error', () => {
-              // HTTP server is not ready yet.
-              process.stdout.write('.');
-            });
-        },
-        2000,
-        ip
-      );
-    })
-    .catch(err => callback(err));
+        const timer = setInterval(
+          ip => {
+            http
+              .get('http://' + ip, res => {
+                const statusCode = res.statusCode;
+                if (statusCode === 200 && waiting) {
+                  waiting = false;
+                  clearTimeout(timer);
+                  // HTTP server is ready.
+                  console.log('Ready!');
+                  callback(null, ip);
+                }
+              })
+              .on('error', () => {
+                // HTTP server is not ready yet.
+                process.stdout.write('.');
+              });
+          },
+          2000,
+          ip
+        );
+      
+    }catch(err){
+      callback(err);
+    }
+  })();
+
 }
 
 // List all VMs and their external IPs in a given zone.
 // callback(error, [[name, ip], [name, ip], ...])
 function listVms(callback) {
-  zone
-    .getVMs()
-    .then(data => {
+  (async () => {
```

Comment:
Same feedback here and below, lets use `async function listVms` instead of the async closure.